fix int warning on large int constants

when we build the c test code (part of testC action) we get:

```
c_out/output.c: In function ‘___bsts_c_IntTest_l_add__sub’:
c_out/output.c:6363:164: warning: integer constant is so large that it is unsigned
 6363 |                                                                     ___bsts_g_Bosatsu_l_Predef_l_eq__Int(___bsts_g_Bosatsu_l_Predef_l_sub(bsts_integer_from_int64(-9223372036854775808),
      |                                                                                                                                                                    ^~~~~~~~~~~~~~~~~~~
c_out/output.c:6379:172: warning: integer constant is so large that it is unsigned
 6379 |                                                                             ___bsts_g_Bosatsu_l_Predef_l_eq__Int(___bsts_g_Bosatsu_l_Predef_l_mul(bsts_integer_from_int64(-9223372036854775808),
      |                                                                                                                                                                            ^~~~~~~~~~~~~~~~~~~
c_out/output.c: In function ‘___bsts_c_PredefTests_l_test__int’:
c_out/output.c:12590:42: warning: integer constant is so large that it is unsigned
12590 |                 bsts_integer_from_int64(-9223372036854775808)));
      |                    
```

This is a code generation error. We should not be generating code that creates a warning. The fix to this is *not* to change any bosatsu code, but to change the ClangGen code to not produce this. We should add a test that any `bsts_interger_from_int64` in the generated code is applied to a constant that won't produce a warning or error in C11.

--- a/core/src/main/scala/dev/bosatsu/codegen/clang/ClangGen.scala
+++ b/core/src/main/scala/dev/bosatsu/codegen/clang/ClangGen.scala
@@ -606,9 +606,12 @@ class ClangGen[K](ns: CompilationNamespace[K]) {
               case _: ArithmeticException =>
                 try {
                   val lv = toBigInteger.longValueExact()
+                  val int64Const =
+                    if (lv == Long.MinValue) Code.Ident("INT64_MIN")
+                    else Code.IntLiteral(BigInt(lv))
                   pv(
                     Code.Ident("bsts_integer_from_int64")(
-                      Code.IntLiteral(BigInt(lv))
+                      int64Const
                     )
                   )
                 } catch {

--- a/core/src/test/scala/dev/bosatsu/codegen/clang/ClangGenTest.scala
+++ b/core/src/test/scala/dev/bosatsu/codegen/clang/ClangGenTest.scala
@@ -725,6 +725,46 @@ main = 4294967296
     }
   }
 
+  test("int64 literal arguments are C11-safe constants") {
+    TestUtils.checkPackageMap("""
+a = -9223372036854775808
+b = -9223372036854775807
+c = 4294967296
+main = a
+""") { pm =>
+      val renderedE = Par.withEC {
+        ClangGen(pm).renderMain(
+          TestUtils.testPackage,
+          Identifier.Name("main"),
+          Code.Ident("run_main")
+        )
+      }
+      renderedE match {
+        case Left(err) =>
+          fail(err.toString)
+        case Right(doc) =>
+          val rendered = doc.render(80)
+          val int64Args =
+            "bsts_integer_from_int64\\(([^)]*)\\)".r
+              .findAllMatchIn(rendered)
+              .map(_.group(1).nn.trim)
+              .toList
+
+          assert(int64Args.nonEmpty)
+          assert(!int64Args.contains("-9223372036854775808"))
+          assert(
+            int64Args.forall { arg =>
+              (arg == "INT64_MIN") || {
+                val parsed = BigInt(arg)
+                (parsed > BigInt(Long.MinValue)) &&
+                (parsed <= BigInt(Long.MaxValue))
+              }
+            }
+          )
+        }
+      }
+    }
+
   test("float literals with sign bit use unsigned bit literals") {
     TestUtils.checkPackageMap("""
 main = -0.0

